packages/web/src/components/OpenSourceDatasets/index.tsx

```diff
@@ -26,6 +35,40 @@ export default function OpenSourceDatasets() {
         );
     }, [dispatch]);
 
+    const openDatasetInApp = (
+        datasetName: string,
+        source: Source,
+        url?: Partial<FileExplorerURLComponents>
+    ) => {
+        dispatch(
+            selection.actions.addQuery({
+                name: `New ${source.name} Query on ${datasetName || "open-source dataset"}`,
+                parts: { ...url, sources: [source] },
+            })
+        );
+        navigate({
+            pathname: "/app",
+            search: `?${currentGlobalURL}`,
+        });
+    };
+
+    const loadDataset = (datasetDetails?: PublicDataset) => {
+        if (!datasetDetails) throw new Error("No dataset provided");
+
+        const dataSourceURL = datasetDetails.path;
+        const { name, extensionGuess } = getNameAndTypeFromSourceUrl(dataSourceURL);
+        const url = datasetDetails?.presetQuery;
+        openDatasetInApp(
+            datasetDetails.name,
+            {
+                name,
+                type: extensionGuess as "csv" | "json" | "parquet",
```

Comment:
What happens if the extension guess is not of type "csv" | "json" | "parquet"?  is that guaranteed? if its guaranteed than why is it a guess?

Reply:
(not written by me so this may be slightly incorrect) this was just an formal typechecking guarantee. 

The function that produced `extensionGuess` (`getNameAndTypeFromSourceUrl`) did only return those options, and defaulted to "csv" if the extension didn't match any of those. However, it wasn't originally written in a way that *technically* guaranteed those return values (since extensionGuess was officially initialized as a generic `string` instead of type `"csv" | "json" | "parquet"`). I've rewritten the function  to explicitly type that return values in [a9141bd](https://github.com/AllenInstitute/biofile-finder/pull/429/commits/a9141bd1ebfff05fad8f3bfb662d7d45039d9e37)

Reply:
It is implicit from the input we allow users to select - not ideal but helps narrow things downstream